S3/S3EventGridPoc/Function.cs

```diff
@@ -0,0 +1,84 @@
+using Amazon.Lambda.Core;
+using Amazon.Lambda.S3Events;
+using Amazon.S3;
+using Amazon.S3.Util;
+using Kusto.Data;
+using Kusto.Data.Net.Client;
+using Kusto.Ingest;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+namespace S3EventGridPoc;
+
+public class Function
+{
+    IAmazonS3 S3Client { get; set; }
+
+    /// <summary>
+    /// Default constructor. This constructor is used by Lambda to construct the instance. When invoked in a Lambda environment
+    /// the AWS credentials will come from the IAM role associated with the function and the AWS region will be set to the
```

Comment:
You say the credentials will come from IAM but you ask for AwsCredentials parameter ... which one is it ?

Reply:
this is also part of the auto generated code. actually I don't need this client